pkgs/os-specific/linux/busybox/sandbox-shell.nix

```diff
@@ -1,9 +1,15 @@
-{ busybox}:
+{ lib, stdenv, busybox, musl }:
 
 # Minimal shell for use as basic /bin/sh in sandbox builds
 busybox.override {
   enableStatic = true;
   enableMinimal = true;
+
+  useMusl =
+    # GNU hosts use glibc which is difficult to statically link against.
+    # Prefer musl if the host platform supports it.
+    stdenv.hostPlatform.isGnu && lib.meta.availableOn stdenv.hostPlatform musl;
```

Comment:
Well, the question is, which choice is better for users?  It sounds like the musl version in, if the glibc one is still much bigger, and there's no other difference between them.

Reply:
Probably not. Though personally I don't mind using `musl` that much because `/bin/sh` is rarely used and it brings the final closure size down.

Imo we don't need to make that decision in this PR.

Reply:
Agree about the comment, but think I would probably still keep `isGnu` here as this is much less of an issue with other `libc`s.

If we switch to `musl` across the board, then we'd probably want to exclude it at least for `bionic`.

Reply:
Happy to switch to `musl` across the board if there is consensus though.

Reply:
bionic is android, right?

Do we even have the sandbox on the other platforms?

Reply:
Android yeah. `busybox-sandbox-shell` is only really used on Linux, so we're just talking about variations of Linux and different cpu architectures here.

Reply:
FWIW, linking against `glibc.static` is not even that bad, the binary is only around 1MB in size. The only issue is it references `glibc` and `libgcc` from the `/nix/store` because of string references to `zoneinfo`, `locale`, `gconv` and `ld.so.cache`.